lib/APMSpan.js

```diff
@@ -21,31 +21,41 @@ import { APMSpanContext } from './APMSpanContext';
 class APMSpan {
     constructor(tracer, fields) {
         this._tracer = tracer;
-        this._spanContext = APMSpan._createSpanContext(fields.childOf);
+        if (fields.childOf && fields.childOf instanceof APMSpan) {
+            fields.childOf = fields.childOf.context();
+        }
+
+        this._initSpanContext(fields.childOf);
 
         this._operationName = fields.operationName;
         this._tags = fields.tags || {};
         this._startMillis = fields.startTime || nowMillis();
         this._endMillis = null;
         this._logs = [];
-        this._sampled = tracer.getSampler().isSampled(this);
 
         this._finished = false;
     }
 
-    static _createSpanContext(parentSpanOrContext) {
+    _initSpanContext(parentSpanContext) {
         const id = generateSpanId();
```

Comment:
I believe this id is being used as the 'interaction id' sent in the header property for outbound requests? If so, then the id needs to be unique per interaction, so needs to be generated at the point where the state is injected.
Otherwise if there are two messages being sent within the same span scope, they will both have the same id.

Although ideally an app developer would place each service invocation in a separate child span, there are no guarantees - atleast if the interaction id is generated at the point of injection, and associated with the Node as an interaction id, then we would end up with a Producer node that has two Interaction based correlation ids, which is supported by the model.